client/src/App.js

```diff
@@ -93,49 +103,127 @@ class App extends Component {
         <p>{this.state.responseToPost}</p>
 
         <div>
-          <button 
-            type='button'
-            onClick={this.handleShowModalClick}>Login</button>
-          {this.state.showModal ? (
-            <Modal onClose={this.handleCloseModal}>
-              <h1>Sign In</h1>
-              <form onSubmit={this.handleModalSubmit}>
-                <label htmlFor="email">
-                  <b>Email</b>
-                </label>
-                <input
-                  type="text"
-                  placeholder="Enter Email"
-                  name="email"
-                  onChange={
-                    e => 
-                    this.setState({
-                      email: e.target.value
-                    })
-                  }
-                />
-                <br />
-                <label htmlFor="psw">
-                  <b>Password</b>
-                </label>
-                <input
-                  type="password"
-                  placeholder="Enter Password"
-                  name="psw"        
-                  onChange={
-                    e => 
-                    this.setState({
-                      password: e.target.value
-                    })
-                  }
-                />
-                <br />
-                <button 
-                type="submit">Submit</button>
-              </form>
-            </Modal>
-          ) : null}
-        </div>
+  <ReactBootstrap.Button type="button" onClick={this.handleShowModalClick}>
```

Comment:
Move the sign-in code into its own class under client/src/signin/signin.js refer to my current branch for an example of how the property table code is outside of App.js for easier debugging

Reply:
Ok -- currently working on it. I'm having a bit of trouble opening/closing the modal in a separate class, but I'll update the branch as soon as I get it working.

Reply:
App.js can control whether the modal is shown or not through the boolean 'showLoginModal'. You can pass in this boolean to your Login class from App.js and also pass in a function that sets it to false, here's a rough idea:

```
App.js
- boolean showLoginModal: true
- function hideLoginModal: this.setState({showLoginModal: false});
- function render(
      <Login showLoginModal hideLoginModal />
)

Login
- if (this.props.showLoginModal) 
      function render (
           ...bunch of cool stuff
            onClickSubmit -> this.props.hideLoginModal
      )
```